src/remote_adapter/remote_adapter_server/main.cc

```diff
@@ -0,0 +1,315 @@
+#include "rpc/this_handler.h"
+
+#include "tcp_message_broker.h"
+#include "utils_manager.h"
+#include "common/constants.h"
+
+using utils_wrappers::UtilsManager;
+
+void PrintUsage() {
+  std::cout << "\nUsage:" << std::endl;
+  std::cout << "------------------------------------------------" << std::endl;
+  std::cout << "For default port usage(port 5555): " << std::endl;
+  std::cout << "./RemoteTestingAdapterServer" << std::endl;
+  std::cout << "------------------------------------------------" << std::endl;
+  std::cout << "For custom port usage: " << std::endl;
+  std::cout << "./RemoteTestingAdapterServer <port>" << std::endl;
+  std::cout << "------------------------------------------------\n"
+            << std::endl;
+  std::cout << "NOTE: Port must be unsigned integer within 1024 - 65535\n";
+}
+
+bool IsUnsignedNumber(const std::string& number) {
+  // Checking for negative numbers
+  if ('-' == number[0] && isdigit(number[1])) {
+    std::cout << "\nNumber is negative!" << std::endl;
+    return false;
+  }
+  // Check that every symbol is a number
+  for (const char& c : number) {
+    if (!isdigit(c))
+      return false;
+  }
+  return true;
+}
+
+void CheckError(const int res) {
+  if (constants::error_codes::SUCCESS == res) {
+    return;
+  }
+  std::string stringified_error;
+  switch (res) {
+    case constants::error_codes::PATH_NOT_FOUND:
+      stringified_error = "Connection not found";
+      break;
+    case constants::error_codes::READ_FAILURE:
+      stringified_error = "Reading failure";
+      break;
+    case constants::error_codes::WRITE_FAILURE:
+      stringified_error = "Writing failure";
+      break;
+    case constants::error_codes::CLOSE_FAILURE:
+      stringified_error = "Closing failure";
+      break;
+    case constants::error_codes::ALREADY_EXISTS:
+      stringified_error = "Channel already exists";
+      break;
+    default:
+      stringified_error = strerror(res);
+      break;
+  }
+  const auto err_obj = std::make_pair(res, stringified_error);
+  rpc::this_handler().respond_error(err_obj);
+}
+
+#ifdef __QNX__
+void print_stack_trace(int pid){
+
+  char out[1024];
+  bt_addr_t pc[16];
+  bt_accessor_t acc;
+  bt_memmap_t memmap;
+
+  memset(&memmap,0,sizeof(bt_memmap_t));
+  bt_init_accessor(&acc, BT_SELF);
+  bt_load_memmap(&acc, &memmap);
+  bt_sprn_memmap(&memmap, out, sizeof(out));
+
+  int cnt = bt_get_backtrace(&acc, pc, sizeof(pc)/sizeof(bt_addr_t));
+  bt_sprnf_addrs(&memmap, pc, cnt, "%a\n", out, sizeof(out), 0);
+  puts(out);
+  
+  bt_unload_memmap(&memmap);
+  bt_release_accessor(&acc);
+}
+
+void segfault_sigaction(int sign, siginfo_t *si, void *arg){ 
+    print_stack_trace(si->si_pid);
+    //** for continue generate core dumped *
+    signal(sign, SIG_DFL);
+    kill(si->si_pid, sign);
+    //**************************************
+}
+#endif
+
+int main(int argc, char* argv[]) {
+
+  UtilsManager::StopApp("AppLinkService");
+  UtilsManager::StopApp("SmartDeviceLink");
```

Comment:
@SNiukalov Are this actions really needed in all use cases of remote server on each platform?

Reply:
@aderiabin Fixed in 1826a1e